hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/AttributeSearchService.kt

```diff
@@ -117,6 +127,28 @@ class AttributeSearchService {
     }
   }
 
+  private fun DateMatcher.validate() {
+    attributeTypes[attribute]
+      ?.also {
+        if (it != AttributeType.DATE) {
+          throw AttributeSearchException("Attribute $attribute is not an date attribute")
+        }
+      }
+      ?: throw AttributeSearchException("Unknown attribute: $attribute")
+
+    if (minValue == null && maxValue == null) {
+      throw AttributeSearchException("Attribute $attribute must have at least 1 min or max value")
+    }
+
+    if (minValue != null && maxValue != null) {
```

Comment:
is it worth adding in future validation for min date and I guess too far in the past validation for max date?